metric/cpu/metrics_windows.go

```diff
@@ -26,13 +26,102 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/elastic/elastic-agent-libs/helpers/windows/pdh"
 	"github.com/elastic/elastic-agent-libs/opt"
 	"github.com/elastic/elastic-agent-system-metrics/metric/system/resolve"
 	"github.com/elastic/gosigar/sys/windows"
 )
 
+var (
+	processorInformationCounter = "\\Processor Information(%s)\\%s"
+	totalKernelTimeCounter      = fmt.Sprintf(processorInformationCounter, "*", "% Privileged Time")
+	totalIdleTimeCounter        = fmt.Sprintf(processorInformationCounter, "*", "% Idle Time")
+	totalUserTimeCounter        = fmt.Sprintf(processorInformationCounter, "*", "% User Time")
+)
+
+var query pdh.Query
+var qError error
+
+func init() {
+	query, qError = buildQuery()
```

Comment:
We are throwing away the error here, is there a better way to initialize this? Perhaps a  or sync.Once or something?

If the counter names are in fact localized, this is where the error would first appear, and we'd never see it with the code the way it is now.

Reply:
sync.Once adds CPU overhead. I'll explore more options.

> If the counter names are in fact localized, this is where the error would first appear, and we'd never see it with the code the way it is now.

I do check for qError at https://github.com/elastic/elastic-agent-system-metrics/blob/35a0aa417f52b238bb1b9f4926465e3b61758e41/metric/cpu/metrics_windows.go#L50-L61.

Reply:
We will pass the error to the caller and eventually to metricbeat.

Reply:
If we have to add an initializer function for system metrics it's not the end of the world. 

Looks like there are a couple of other instances of this problem in here already, like https://github.com/elastic/elastic-agent-system-metrics/blob/5a3873d8d6f807e2c111b437c10291a13d9c2b83/report/metrics_common.go#L38-L47

Although this is the first one where we have a real reason for it to fail (thanks Windows localization).

Reply:
Please don't do this in `init()`, if there are any delays or blocking look ups that can delay startup of elastic-agent.  That is a problem for things like running as a Windows service where we have a set amount of time to respond to the Service Manager.

Reply:
@leehinman I've opened another, smaller PR https://github.com/elastic/elastic-agent-system-metrics/pull/194 just for handling this. Can you take a look?

Reply:
✅ That is a big improvement.  Thank You.